### ISSUE TITLE
Joe Weekly Journals

### ISSUE THREAD

**Issue by Jlandau2** [YCPRadioTelescope/RT-Contracts#23]
Week 10/9 - 10/16
This week our AWS progress was entirely stunted. An individual who shall not be named caused our AWS access to be blocked, due to extreme EC2 instance creation. Certain roles and security groups were deleted, causing us to have to terminate the dev and prod environments. Essentially, we are back at square zero with AWS progress. I believe we will be redefining our milestone 1 goals to just the dev environment functioning, with the prod environment existing but potentially not hooked up to the domain. At the time of writing this, we have no ability to create EC2 instances, so my hope is to regain access, which the individual who messed up is trying to fix, and create the dev environment immediately for Venus.

I also reviewed compatibility documentation. Windows server 2019 seems to not be compatible with our hardware. I contacted IT about licensing for educational purposes, and they seem confident about our use case. However, they are going to contact Microsoft to double check. In the meantime, I have installed Windows Server 2016 Standard Evaluation. This comes with a 180 Day trial period. This is plenty of time for us to figure out our preliminary data-syncing concept, as well become acclimated to the Windows Server environment.

https://github.com/YCPRadioTelescope/RT-Contracts/issues/22

I also created a few weather data tables and controllers for Team Jupiter. However I am unable to continue on the RF data table because their request is unclear.

https://github.com/YCPRadioTelescope/RT-Contracts/issues/19
https://github.com/YCPRadioTelescope/RT-Contracts/issues/20
https://github.com/YCPRadioTelescope/RT-Contracts/issues/21

I have not made any commits, as I need to implement tests. Mostly I just was focusing on the other tasks, as they seemed of higher priority.

**Comment by Jlandau2** [comment]
Week 10/16-10/23
This week I made substantial progress on setting up the server. Todd was able to give us the Windows Server 2016 StandardServer edition. So we resolved our OS licensing.  However, now I have run into yet another licensing issue. We have 120 days  (now 118) of free Remote Desktop Services. After that, we will be required to purchase Client Access Licenses for each individual who wants to remotely access the server. Todd has access to multiple Windows Server 2016 licenses through MSDN, so we could interchange them. The OS license changing script is relatively easy, so it could save them a lot of money for at least a year.

On top of this, I also got our backend application running on the server. IntelliJ, MySQL, and all of the supporting languages have been properly installed. This was my big goal for milestone 1. 

https://github.com/YCPRadioTelescope/RT-Contracts/issues/25

On top of getting the backend running, I also delved into getting the Remote Desktop service working. So far I have created and Active Directory full of 50 users, each with their own randomized password. My goal is to have a request sheet on google drive for people to request a User account, which I can then share with them. User names have been kept generic, "RTUserN" with N being the number of the account given to them, which will be handed out sequentially. Todd, Kerry, and Hake have accounts named after them, as they will remain a constant into the next year. However, myself followed by whoever else requests one will be given generic accounts. By using numbers instead of names, we can forego re-instantiating accounts when the semester or year changes. This will also make licensing easier, as you can license by user, so we can issue the licenses to the generically named accounts without having to create the user from scratch and then add the license. 

For the users to connect, I have instantiated a domain controller. After a discussion with the networking department, I have found that this is how we can connect individuals on campus to our server. However, because of the York College domain structure, we will have to connect anyone who wants to view it off campus through a vpn to a middle server (potentially AWS) which has a vpn connected to our server here on campus. For now I am focused on getting the teams on campus connected, as I have to still figure out how to utilize the AWS server to actually facilitate the connection to my domain. In fact, I am still figuring out how to get individuals to connect to our domain. Currently you can log in, but it gives a "Sign-in method not authorized" error. I have to still add acceptable programs to the session-based remote desktop collections. This is how you prevent individuals from remoting in and accessing everything. So for instance the Server Manager will not be available to any of the teams but myself.  I would like to create group policies for each team, primarily so that Joel, Todd and the other admins and myself can remote in and control everything while the other teams do not.  

https://github.com/YCPRadioTelescope/RT-Contracts/issues/26

I have no commits as I did not touch the application this week.

**Comment by Jlandau2** [comment]
This week was relatively quiet. AWS access was restored, so I helped to stand up the RDS for development and production, and Ed and Joel have both been able to regain application connection to it. Other than that I have stepped back to local server stuff to make more progress on remote desktop. However, I made the mistake of changing my home pc's OS and accidentally deleted the spreadsheet I had of all the passwords. So I had to pretty much start the Remote Desktop infrastructure from the beginning, which was much faster, and I was able to double check my documentation on it. For future planning, I will not keep the password table so detached, just on my laptop. I need to figure out how the active directory logs the new passwords they will have when they log in the first time, which is a feature I have enabled. I also need to figure out how to password requirements, so nobody has a "password" password.

So my goals for next week are as follows:

1. Create the session collections, which determines what apps the remote connections will access
2. Create a local connection between a computer on campus and my server
3. Create a script that adds a timestamp and trigger to all tables in the db 

https://github.com/YCPRadioTelescope/RT-Contracts/issues/26
